scripts/docker/build_docker.py

```diff
@@ -80,6 +80,10 @@ def __init__(self, args_list):
                                               action = 'store_true',
                                               help   = 'also update \"latest\" tag in remote docker repo(s)')
 
+        docker_remote_args_group.add_argument('--dry-run',
+                                              action = 'store_true',
+                                              help   = 'If given, the script will not push the built images to gcr.')
```

Comment:
Would it be easier to make `--gcr-project` optional? I believe the only difference will be the removal of `--squash` from the `docker` command.

Reply:
Sure, I can make `--gcr-project` optional instead of `--dry-run`. Would you mind please elaborate on `--squash`? should that be ignored if `--gcr-project` is not given?

Reply:
You can read about it [here](https://docs.docker.com/engine/reference/commandline/build/#squash-an-images-layers---squash-experimental) - I'm not sure we actually need to be doing this in the first place, but it will not affect the functionality of the image at all.

Reply:
Thank you for the link about `--squash`, actually I was wondering if you were suggesting using this flag to achieve functionality similar to `--dry-run`. 

> 'm not sure we actually need to be doing this in the first place,

Reply:
`--squash`

Reply:
We discussed offline, but we do want to keep `--squash` because it can reduce final image size in some cases.

Reply:
I guess `--gcr-project` is optional by default, I just updated its help message to reflect images will not be pushed if a value for the argument is not provided. I agree leveraging this argument is simpler/clearer than a new argument.